src/test/java/teammates/test/cases/browsertests/FeedbackMsqQuestionUiTest.java

```diff
@@ -216,6 +216,182 @@ public void testEditQuestionAction() throws Exception {
                 Const.ParamsNames.FEEDBACK_QUESTION_GENERATEDOPTIONS + "-1",
                 FeedbackParticipantType.TEAMS.toString());
 
+        ______TS("MSQ: min/max selectable options");
+        feedbackEditPage.clickSaveExistingQuestionButton(1);
+        feedbackEditPage.verifyStatus(Const.StatusMessages.FEEDBACK_QUESTION_EDITED);
+
+        // Check min/max selectable restrictions for
+        // new MSQ question with custom options
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.selectNewQuestionType("MSQ");
+        feedbackEditPage.fillQuestionTextBoxForNewQuestion("Custom options");
+        feedbackEditPage.fillMsqOptionForNewQuestion(0, "A");
+        feedbackEditPage.fillMsqOptionForNewQuestion(1, "B");
+        feedbackEditPage.clickAddMoreMsqOptionLinkForNewQuestion();
+        feedbackEditPage.clickAddMoreMsqOptionLinkForNewQuestion();
+        feedbackEditPage.fillMsqOptionForNewQuestion(2, "C");
+        feedbackEditPage.fillMsqOptionForNewQuestion(3, "D");
+        checkMinMaxSelectableRestrictionForCustomOptions(-1);
+
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.verifyStatus(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+
+        // Check min/max selectable restrictions for
+        // existing MSQ question with custom options
+        feedbackEditPage.clickEditQuestionButton(2);
+        checkMinMaxSelectableRestrictionForCustomOptions(2);
+        feedbackEditPage.clickDeleteQuestionLink(2);
+        feedbackEditPage.waitForConfirmationModalAndClickOk();
+        feedbackEditPage.verifyStatus(Const.StatusMessages.FEEDBACK_QUESTION_DELETED);
+
+        // Check min/max selectable restrictions for new
+        // MSQ question with options generated from students
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.selectNewQuestionType("MSQ");
+        feedbackEditPage.fillQuestionTextBoxForNewQuestion("Msq generated options");
+        feedbackEditPage.clickGenerateMsqOptionsCheckbox(-1);
+        checkMinMaxSelectableRestrictionsForAllGenerateOptionSelections(-1);
+
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.verifyStatus(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+
+        // Check min/max selectable restrictions for existing
+        // MSQ question with options generated from students
+        feedbackEditPage.clickEditQuestionButton(2);
+        checkMinMaxSelectableRestrictionsForAllGenerateOptionSelections(2);
+        feedbackEditPage.clickDeleteQuestionLink(1);
+        feedbackEditPage.waitForConfirmationModalAndClickOk();
+        feedbackEditPage.verifyStatus(Const.StatusMessages.FEEDBACK_QUESTION_DELETED);
+    }
+
+    private void checkMinMaxSelectableRestrictionForCustomOptions(int qnNumber) {
```

Comment:
This method is filled with magic numbers (e.g. 5, 6, and especially `4`). The way I understand it, it should be the method parameter and determined by this method's call sites.